### DISCUSSION TITLE
TEAC Head load solenoid

### DISCUSSION THREAD

**Discussion by barbeque** [keirf/greaseweazle#207] [Q&A]
I want to write a 180K diskette (for a single-sided TEAC FD-54A in a Sanyo MBC-555) and I have a 360K disk drive (a TEAC FD-55B-06-U, out of an NEC PC-8801mkII.) I'm working with an F7 board ("GW_F730" written on it) and a V4 board.

When using the `gw` utility, it seems to write the image without errors – the drive is selected, the head progresses towards the centre.

However, when I read back the disk and look at it in the HxC utility, it just looks like random noise, like the disk was never read:

![Screen Shot 2022-07-10 at 9 10 20 PM](https://user-images.githubusercontent.com/423811/178181604-b4b35fe3-a020-4b33-a22c-0808a6a3f283.png)

I ran into a similar problem last year with an FD-55GFV-70 out of a PC-8801MH. I picked through the TEAC data sheet for the drives (a little tricky to find the exact right one for these drives, and they don't describe the strapping or configuration.)

It seems like a handful of TEAC drives were shipped with a head-load solenoid that needs to be actively turned on in order for them to work, and it makes sense to me that if you didn't do this, you would just be reading/writing cosmic rays or something instead of the disk.

> The head load mechanism functions to make the head in contact with the disk when only required so as to reduce disk and magnetic head wear. This mechanism is constructed with head load solenoid, arm lifter, etc. For a single sided FDD, the head pad attached to the pad arm of the head carriage is depressed against the SIDE 1 of the disk with appropriate pressure and the disk is held with head and head pad. For a double sided FDD, the SIDE 1 head is depressed with appropriate pressure instead of the head pad, as well as the single sided FDD.

Indeed, if I take the top off the drive and look in, when I insert the disk into the drive, the head doesn't descend all the way to meet it. There's a little spring-loaded arm on the side holding it _just_ off the surface of the disk.

The specific mechanism for determining the output of the READY pin seems to be related to this solenoid: https://archive.org/details/Teac_FD-55_Drive_Maintenance_Manual_1983_Teac_Corporation_of_America/page/n23/mode/2up

![Screen Shot 2022-07-10 at 9 50 41 PM](https://user-images.githubusercontent.com/423811/178185166-2b2c0c58-7aaa-4326-8f06-d22bf52204dc.png)

I think the real FDC is supposed to drive the head load state through pin 4 (as per [this TEAC spec document](http://www.bitsavers.org/pdf/teac/FD55_Spec.pdf)), although I have not been able to get a logic analyzer capture from a working machine. From the calibration and diagnostic information in the back of the above linked Radio Shack manual, it seems like technicians would have a special piece of equipment ("Simulator KA") that can drive head load when they want it.

I tried setting pin state from `gw set pin 4 L` on my F7 but I got an "invalid pin" error, I am guessing because my GW hardware was not able to drive that pin from the microcontroller.

So I went to AmigaKit and got a V4 board, which _can_ set pin 4, but either high or low does not seem to make the actuator move when either writing or reading.

My guess then is that this is something to do with timing, since the above timing chart seems to imply that the head load enable kicks in later, so maybe setting it before turning the motor on is being ignored by the drive's logic.

So my questions are:
 - Am I sane in concluding this is the secret to getting one of these drives to work, or do I need to do more research?
 - Is it possible to drive this pin from the Greaseweazle?

I don't mind having to get another drive, but I do like these little 360K guys. I might try playing with jumpers later, as the configuration is as-shipped by NEC with the "HL" strap unset (only DS0 and SM are bridged, and the lower pin of DS1 is connected to the lower pin of MH.)

Thanks for any advice.

**Comment by keirf** [comment]
It makes sense F7 could not set pin 4, if yours is an F7 v1 (it has the MCU mounted on the PCB at 45 degree angle).

1. I've not had a drive that needs head loading myself before. Usually it is strappable (as it is on the FD-55 you linked to).
2. You can certainly drive pin 4 from a GW v4. You can check with a multimeter in volts mode (although you may need to add a 1k resistor from pin 4 to ground, as the output is open collector).

The other common use of pin 4 is to drive the in-use lamp on the drive bezel. So you might want to see if that lights in response to pin 4. Also you might want to look at the PCB and see if pin 4 is connected anywhere at all!

**Reply by barbeque** [reply]
Thanks for the quick reply. The light does illuminate, but maybe the in-use logic that drives it works differently when the drive is equipped with the solenoid. 

I’ll see if the ready pin is set.

**Reply by barbeque** [reply]
Small update: I've made some progress setting the "HS" strap near the drive select pins: when "SM" is also set, it makes the head solenoid trigger and drop the head as the drive spins up – according to [this list of jumpers](https://retrocmp.de/fdd/teac/TEAC_FD-55_Jumper-settings.pdf) that I somehow missed over the last few days of searching, that means the head solenoid is being triggered by drive select – and the LED is driven by drive select as well.

However, despite the head dropping now, a write-and-readback still seems to be an empty disk, so there is probably something else required.

**Reply by barbeque** [reply]
Long overdue update: Reading the disk back as SCP instead of HFE made the disk appear in the HxC tools, I'm not entirely sure why. But it does work! I was trying to write a Sanyo MBC-555 (reportedly FM encoded) MSDOS 1.2 diskette using the PC-8801mkII's TEAC floppy drive.

I've also had to strap an NEC FD-1155's "head load" jumper on a PC-9801RA2 (move the "USE" jumper to position 2) to write disks and control the drive for that computer.
